stop_and_wait_plugin/include/stop_and_wait_plugin.h

```diff
@@ -31,174 +31,93 @@
 #include <lanelet2_core/geometry/LineString.h>
 #include <geometry_msgs/PoseStamped.h>
 #include <geometry_msgs/TwistStamped.h>
-
+#include "stop_and_wait_config.h"
 
 namespace stop_and_wait_plugin
 {
-    /**
-     * \brief Convenience class for pairing 2d points with speeds
-     */ 
-    struct PointSpeedPair
-    {
-        lanelet::BasicPoint2d point;
-        double speed=0;
-    };
-
-    class StopandWait
-    {
-    public:
-        /** 
-         * \brief Constructor
-         */
-        StopandWait(){};
-
-          /**
-         * \brief Service callback for trajectory planning
-         * 
-         * \param req The service request
-         * \param resp The service response
-         * 
-         * \return True if success. False otherwise
-         */ 
-        bool plan_trajectory_cb(cav_srvs::PlanTrajectoryRequest& req, cav_srvs::PlanTrajectoryResponse& resp);
-        
-        /**
-         * \brief General entry point to begin the operation of this class
-         */
-        void run();
-
-        /**
-         * \brief Converts a set of requested STOP_AND_WAIT maneuvers to point speed limit pairs. 
-         * 
-         * \param maneuvers The list of maneuvers to convert
-         * \param max_starting_downtrack The maximum downtrack that is allowed for the first maneuver. This should be set to the vehicle position or earlier.
-         *                               If the first maneuver exceeds this then it's downtrack will be shifted to this value.
-         * 
-         * 
-         * \return List of centerline points paired with speed limits
-         */ 
-        std::vector<PointSpeedPair> maneuvers_to_points(const std::vector<cav_msgs::Maneuver>& maneuvers,
-                                                                      double starting_downtrack,
-                                                                      const carma_wm::WorldModelConstPtr& wm, const cav_msgs::VehicleState& state);
-          /**
-         * \brief Method converts a list of lanelet centerline points and current vehicle state into a usable list of trajectory points for trajectory planning
-         * 
-         * \param points The set of points that define the current lane the vehicle is in and are defined based on the request planning maneuvers. 
-         *               These points must be in the same lane as the vehicle and must extend in front of it though it is fine if they also extend behind it. 
-         * \param state The current state of the vehicle
-         * 
-         * \return A list of trajectory points to send to the carma planning stack
-         */ 
-        std::vector<cav_msgs::TrajectoryPlanPoint> compose_trajectory_from_centerline(
-        const std::vector<PointSpeedPair>& points, const cav_msgs::VehicleState& state);
+using PublishPluginDiscoveryCB = std::function<void(const cav_msgs::Plugin&)>;
+/**
+ * \brief Convenience class for pairing 2d points with speeds
+ */
+struct PointSpeedPair
```

Comment:
How so, can you explain?

Reply:
This struct has 3 members now and is different from the point speed pair struct being used in the other plugins.
Copying a link to the definition https://github.com/usdot-fhwa-stol/carma-platform/blob/5bfa0997428c19d423d5d07823fff30987c13461/stop_and_wait_plugin/include/stop_and_wait_plugin.h#L42

Reply:
I'll change the name since it does indeed have 3 members (therefore not a pair). However, the comparison with other plugins is not a concern regardless since this is a class under a different namespace it is not required to have any equivalent with external packages.

Reply:
Per our discussion it looks like lanelet::Id is not used, so I will remove it and leave the name as is for now.